src/DriveShare.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace Owncloud\OcisPhpSdk;
+
+use OpenAPI\Client\Model\Permission as ApiPermission;
+use Owncloud\OcisPhpSdk\Exception\InvalidResponseException;
+
+class DriveShare
+{
+    private ApiPermission $apiPermission;
+    private string $driveId;
+
+
+    /**
+     * @ignore The developer using the SDK does not need to create share objects manually,
+     *         but should use the OcisResource class to invite people to a resource and
+     *         that will create ShareCreated objects
+     */
+    public function __construct(
+        ApiPermission $apiPermission,
+        string        $driveId
+    ) {
+        $this->apiPermission = $apiPermission;
+        $this->driveId = $driveId;
+    }
+
+    public function getPermissionId(): string
+    {
+        $id = $this->apiPermission->getId();
+        if ($id === null || $id === '') {
+            throw new InvalidResponseException(
+                "Invalid id returned for permission '" . print_r($id, true) . "'"
+            );
+        }
+        return (string)$id;
```

Comment:
```suggestion
        return $id;
```

`getId` always returns `string` or `null` and we handled `null` above. So we already know that `$id` is a string.